### PR TITLE
config skipreasons

### PR DESCRIPTION
**Pull request by jspaaks**
**List of related issues or pull requests**

Refs: #179 #165

Includes PR #171

**Describe the changes made in this pull request**

Changes the configuration parameter names and their types. They are now either None (`~` in YAML) or `str` if the keyname is included in `.howfairis.yml`, or the key name is omitted entirely (or commented out). Their new names and default configuration are 
```yaml
##   Uncomment a line if you want to skip a given category of checks

# skip_repository_checks_reason: <reason for skipping goes here>
# skip_license_checks_reason: <reason for skipping goes here>
# skip_registry_checks_reason: <reason for skipping goes here>
# skip_citation_checks_reason: <reason for skipping goes here>
# skip_checklist_checks_reason: <reason for skipping goes here>

include_comments: false
```

The schema has been updated accordingly.

Three behaviors are possible:
1. run checks like normal, this is triggered when
    - the effective configuration does not include the corresponding `skip_<recommendation>_checks_reason` key
    - the effective configuration includes the corresponding `skip_<recommendation>_checks_reason` key but it has no value
    - the effective configuration includes the corresponding `skip_<recommendation>_checks_reason` key but its value is `~`
2. make one category True, print the reason. This behavior is triggered when
    - the effective configuration includes the corresponding `skip_<recommendation>_checks_reason` key, its type is `str`, and the string is not empty
3. make one category True, print that there was no reason given
    - the effective configuration includes the corresponding `skip_<recommendation>_checks_reason` key, its type is `str`, but the string is empty

**Instructions to review the pull request**

Make a local config file by `howfairis -d > .custom.howfairis.yml` and subsequent editing, e.g.

```yaml
##   Uncomment a line if you want to skip a given category of checks

# skip_repository_checks_reason: <reason for skipping goes here>
# skip_license_checks_reason: <reason for skipping goes here>
skip_registry_checks_reason: "I just don't feel like it"
skip_citation_checks_reason: ~
skip_checklist_checks_reason: ""

include_comments: false
```

Running `howfairis` with user config (`-c` flag) should return something like:

```
$ howfairis https://github.com/fair-software/howfairis -c .custom-howfairis.yml 
Checking compliance with fair-software.eu...
url: https://github.com/fair-software/howfairis
Local configuration file: .custom-howfairis.yml
(1/5) repository:
      ✓ has_open_repository
(2/5) license:
      ✓ has_license
(3/5) registry:
      ✓ skipped (reason: I just don't feel like it)
(4/5) citation:
      × has_citation_file
      ✓ has_citationcff_file
      × has_codemeta_file
      ✓ has_zenodo_badge
      ✓ has_zenodo_metadata_file
(5/5) checklist:
      ✓ skipped (no reason provided)

Calculated compliance: ● ● ● ● ●

Expected badge is equal to the actual badge. It's all good.
```

Still needs documentation updates

### CODE REVIEW

File: howfairis/mixins/checklist_mixin.py [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+        print("(5/5) checklist:")
+        reason = self.skip_checklist_checks_reason
+        if isinstance(reason, str):
+            if reason == "":
+                self._print_state(check_name="skipped (no reason provided)", state=True)
+            else:
+                self._print_state(check_name="skipped (reason: {0})".format(reason), state=True)
+            results = [True]
+        if reason is None:
+            results = [self.has_core_infrastructures_badge()]
```
Comment (sverhoeven) [inline howfairis/mixins/checklist_mixin.py:13]: As config has already been passed through schema validation its either str or None. Can code be more 
```suggestion
        if reason is None:
            results = [self.has_core_infrastructures_badge()]
        else:
            if reason == "":
                self._print_state(check_name="skipped (no reason provided)", state=True)
            else:
                self._print_state(check_name="skipped (reason: {0})".format(reason), state=True)
            results = [True]
```

**Review by sverhoeven — COMMENTED** [review]
The https://github.com/fair-software/badge-test repo still uses old format, can it be updated.

If I only remove `#` from example config I get following error a 
```
ruamel.yaml.composer.ComposerError: expected a single document in the stream
  in "<unicode string>", line 3, column 2:
     skip_repository_checks_reason: < ... 
     ^ (line: 3)
but found another document
  in "<unicode string>", line 9, column 1:
    include_comments: false
    ^ (line: 9)
```
I need to remove '# ` for it to work, maybe remove the extra space so that does not happen.

File: howfairis/mixins/checklist_mixin.py (hunk 1 above)
Comment (jspaaks) [reply howfairis/mixins/checklist_mixin.py]: updated in subsequent commits

**Review by fdiblen — APPROVED** [review]
To me it is good to merge. :merman:

**Comment by jspaaks** [comment]
Thanks for reviewing!

**Pull request merged**